{{cookiecutter.project_name}}/default.nix

```diff
@@ -7,31 +7,38 @@ let
   gitignore = pkgs.nix-gitignore.gitignoreSourcePure [ ./.gitignore ];
 
   myHaskellPackages = pkgs.haskell.packages.${compiler}.override {
-    overrides = se: su: {
+    overrides = hself: hsuper: {
       "{{cookiecutter.project_name}}" =
-        se.callCabal2nix
+        hself.callCabal2nix
           "{{cookiecutter.project_name}}"
           (gitignore ./.)
           {};
     };
   };
 
-in
-rec
-{
-  "{{cookiecutter.project_name}}" = myHaskellPackages."{{cookiecutter.project_name}}";
   shell = myHaskellPackages.shellFor {
     packages = p: [
       p."{{cookiecutter.project_name}}"
     ];
     buildInputs = with pkgs.haskellPackages; [
-      cabal-install
+      myHaskellPackages.cabal-install
       ghcid
       ormolu
       hlint
-      pkgs.niv
+      (import sources.niv {}).niv
       pkgs.nixpkgs-fmt
     ];
     withHoogle = true;
   };
-}
+
+  exe = pkgs.haskell.lib.justStaticExecutables (myHaskellPackages."{{cookiecutter.project_name}}");
+
+  docker = pkgs.dockerTools.buildImage {
+    name = "{{cookiecutter.project_name}}";
+    config.Cmd = [ "${exe}/bin/{{cookiecutter.project_name}}" ];
+  };
+in
+  if pkgs.lib.inNixShell then shell else {
```

Comment:
I'm not sure if this is a good idea. Maybe I'm in an (possibly unrelated) nix-shell, and trying to `nix-build` the project. I'd build the `shell` attribute instead, which is not what I want. If we can use a similar function meaning "this expression is invoked via `nix-shell`", that's fine; but in this case I'd prefer having a `shell.nix`.

Reply:
You are right. I re-created the `shell.nix` file and exported a `shell` attribute like before.